framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -635,8 +640,13 @@ private void handleUninferredTypeVariables(
                 final AnnotatedTypeMirror inferredType = inferredArgs.get(typeVar);
 
                 if (inferredType == null) {
-                    AnnotatedTypeMirror dummy = typeFactory.getUninferredWildcardType(atv);
-                    inferredArgs.put(atv.getUnderlyingType(), dummy);
+                    AnnotatedTypeMirror dummy = atv.getUpperBound().deepCopy();
```

Comment:
The upper bound of the type variable might refer to other type variables. It seems wrong to directly use the upper bound as inferred argument. Why is this correct?